internal/stategen/cleanup.go

```diff
@@ -0,0 +1,225 @@
+// a given module from the internal/postprocessor/config.yaml file.
+// NOTE: This function does not remove modules from manual-clients.
+func cleanupPostProcessorConfig(repoRoot, moduleName string) error {
+	configPath := filepath.Join(repoRoot, "internal", "postprocessor", "config.yaml")
+	fileBytes, err := os.ReadFile(configPath)
+	if err != nil {
+		return err
+	}
+
+	lines := strings.Split(string(fileBytes), "\n")
+	var newLines []string
+
+	// First pass: remove from modules list.
+	inModules := false
+	for _, line := range lines {
+		if strings.HasPrefix(line, "modules:") {
+			inModules = true
+		} else if !strings.HasPrefix(line, " ") { // top-level key
+			inModules = false
+		}
+		if inModules && strings.TrimSpace(line) == "- "+moduleName {
+			continue
+		}
+		newLines = append(newLines, line)
+	}
+
+	// Second pass: remove from service-configs.
+	lines = newLines
+	newLines = []string{}
+	inServiceConfigs := false
+	var serviceConfigBlock []string
+	isTargetServiceConfig := false
+	for _, line := range lines {
+		if strings.HasPrefix(line, "service-configs:") {
+			inServiceConfigs = true
+			newLines = append(newLines, line)
+			continue
+		}
+		if !strings.HasPrefix(line, " ") { // top-level key
+			if inServiceConfigs {
+				if !isTargetServiceConfig {
+					newLines = append(newLines, serviceConfigBlock...)
+				}
+				isTargetServiceConfig = false
+				serviceConfigBlock = nil
+			}
+			inServiceConfigs = false
+		}
+
+		if !inServiceConfigs {
+			newLines = append(newLines, line)
+			continue
+		}
+
+		// In service-configs section
+		if strings.HasPrefix(line, "  - input-directory:") {
+			if !isTargetServiceConfig {
+				newLines = append(newLines, serviceConfigBlock...)
+			}
+			serviceConfigBlock = []string{line}
+			isTargetServiceConfig = false
+		} else {
+			serviceConfigBlock = append(serviceConfigBlock, line)
+		}
+
+		if strings.Contains(line, "import-path:") {
+			path := strings.TrimSpace(strings.Split(line, ":")[1])
+			if strings.HasPrefix(path, "cloud.google.com/go/"+moduleName+"/") {
+				isTargetServiceConfig = true
+			}
+		}
+	}
+	// flush last block
+	if inServiceConfigs && !isTargetServiceConfig {
+		newLines = append(newLines, serviceConfigBlock...)
+	}
+
+	output := strings.Join(newLines, "\n")
+	if bytes.Equal([]byte(output), fileBytes) {
+		return nil
+	}
+
+	return os.WriteFile(configPath, []byte(output), 0644)
+}
+
+// cleanupOwlBotYaml removes entries for a given module from the
+// .github/.OwlBot.yaml file.
+func cleanupOwlBotYaml(repoRoot, moduleName string) error {
+	owlBotPath := filepath.Join(repoRoot, ".github", ".OwlBot.yaml")
+	fileBytes, err := os.ReadFile(owlBotPath)
+	if err != nil {
+		return err
+	}
+
+	lines := strings.Split(string(fileBytes), "\n")
+	var newLines []string
+	for i := 0; i < len(lines); i++ {
+		line := lines[i]
+		if strings.Contains(line, "source:") {
+			// It's a source line, check it for the module name.
+			if strings.Contains(line, moduleName) {
```

Comment:
I think we should add a slash here - so that the "ai" module doesn't end up stripping "aiplatform" entries, and ditto "pubsub" and "pubsublite".

Reply:
Good catch! I fixed this and tested it manually against `ai`. The resulting diffs LGTM.